modules/services/aerospace/default.nix

```diff
@@ -0,0 +1,156 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.aerospace;
+
+  format = pkgs.formats.toml { };
+  configFile = format.generate "aerospace.toml" cfg.settings;
+in
+
+{
+  options = with types; {
+    services.aerospace = {
+      enable = mkEnableOption "AeroSpace window manager";
+
+      package = mkOption {
+        type = types.path;
+        default = pkgs.aerospace;
+        description = "The AeroSpace package to use.";
+      };
+
+      settings = mkOption {
+        type = submodule {
+          freeformType = format.type;
+          options = {
+            start-at-login = mkOption {
+              type = addCheck bool (b: !false || !cfg.enable);
+              default = false;
+              description = "Do not start AeroSpace at login. (Managed by launchd instead)";
+            };
+            after-login-command = mkOption {
+              type = addCheck (listOf str) (l: l == [ ] || !cfg.enable);
+              default = [ ];
+              description = "Do not use AeroSpace to run commands after login. (Managed by launchd instead)";
+            };
+            after-startup-command = mkOption {
+              type = addCheck (listOf str) (l: l == [ ] || !cfg.enable);
+              default = [ ];
+              description = "Do not use AeroSpace to run commands after startup. (Managed by launchd instead)";
+            };
```

Comment:
Why do we want to prevent users from configuring `aerospace` to run commands after logging in?

Reply:
These commands only run if aerospace starts itself (via the `start-at-login` option), as opposed to via `launchd`